script_test.go

```diff
@@ -60,16 +62,48 @@ func TestScripts(t *testing.T) {
 		}
 		t.Fatalf("failed to get GOPATH for test: %v\n%s", err, stderr)
 	}
-	modTestGOPATH = string(out)
+	modTestGOPATH = strings.TrimSpace(string(out))
+
+	ucd, err := os.UserCacheDir()
+	if err != nil {
+		t.Fatalf("failed to get os.UserCacheDir: %v", err)
+	}
 
 	p := testscript.Params{
 		Dir: "testdata",
 		Setup: func(e *testscript.Env) error {
+			// TODO feels like this could be a method on testscript.Env?
+			getEnv := func(s string) string {
+				cmp := s + "="
+				for i := len(e.Vars) - 1; i >= 0; i-- {
+					v := e.Vars[i]
+					if strings.HasPrefix(v, cmp) {
+						return strings.TrimPrefix(v, cmp)
```

Comment:
nit, but you could avoid one `strings` call, like:

```
if val := strings.TrimPrefix(v, cmp); val != v {
    return val
}
```

Reply:
I feel like this is one of those situations where there isn't a precise answer, computationally, in terms of which is better. Largely because it depends on the inputs you're seeing. Hence I would tend to prefer something slightly more readable. I'm assuming your point is about computational cost?

I would roughly guess the call to `strings.HasPrefix` is inlined in any case?